### DISCUSSION TITLE
【萌新求助】antd Tree组件可以判断某个TreeNode是否可见吗

### DISCUSSION THREAD

**Discussion by dzqdzq** [ant-design/ant-design#45094] [Q&A] [answered]
最近在搞一个vscode插件，功能是显示prefab的节点树， 左边显示节点树，右边显示属性面板。   属性面板里有id属性可以点击，点击的效果是对应节点树里的某个节点突出显示0.5秒然后回归正常。目前已经实现了这个。
碰到了一个问题，
1. 要突出展示的节点可能不可见。不可见的原因是因为树节点层次比较多，一屏幕展示不下。（ 期望：节点在屏幕内可见）
![image](https://github.com/ant-design/ant-design/assets/18027587/50672cb8-af2c-4a34-bfbe-766599807ab8)

项目仓库： https://github.com/dzqdzq/prefabWebUI

**Comment by dzqdzq** [comment, marked as answer]
经过了这么天，没有一个回复，有点失落， 我不得不放弃antd而使用[react-sortable-tree](https://github.com/frontend-collective/react-sortable-tree)了，虽然比较low，但是功能能满足

**Comment by yoyo837** [comment]
scrollToView?

**Comment by BoyYangzai** [comment]
好久没看 antd 了 刚好邮箱提示看到了 回复一下
首先是对这个业务场景
  期望：节点在屏幕内可见 可以通过一个 hook 来判断 当你点击右侧 id 的时候 左侧对应的对应的节点是否在可视窗口内即可 
  某个 dom 元素是否在可视窗口是个很常见的需求 或者 就像楼上 yoyo 说的 可以直接滚动到对应的 dom 节点上去 也是一种变相的处理方式
其次是想说 不要失落哈哈哈 discussion 本来看的人就比较少 可能都没有人看到这个 discussion 或者 看到了可能没有 get 到你的点
你看刚刚回复 我和yoyo刚好看到了 所以就可以及时回复了
所以 保持好心态(*^▽^*)

**Reply by dzqdzq** [reply]
首先我没有看到有scrollToBottom    scrollToTop    scrollToItem  类似这样的方法。    其实对于使用者来说，只需要selectKey这样的方法，底层应该自动滚动或者移动对应item到可视区域。    而不是我选择了某个key,   没有任何响应， 还以为这个key不存在， 我知道antd其实响应了，只是看不到而已。

**Reply by dzqdzq** [reply]
你们可以参考下[react-sortable-tree](https://github.com/frontend-collective/react-sortable-tree)这个库，功能算是非常完善了，它搜索某个节点的时候，这个节点如果不可见，会自动移动(滚动)这个树。   它的搜索的时候并不会删除(隐藏)非相关节点，而是高亮显示，这一点和antd Tree 还不一样， antd Tree的实现其实也挺好，  不过可以让用户选择就更好了， 搜索不隐藏非相关节点，而是突出显示相关节点。 两者都支持最好不过。 我个人还是比较喜欢antd的，ui设计的很棒，希望越来越好，功能越来越完善

**Comment by yoyo837** [comment]
没有办法保证每个GitHub的提醒都及时看到, 也没有办法保证回答和解决所有问题. 我个人算是看消息比较积极的了, 当前 discussion 也确实首次看到, 邮件一提醒我就立马点进来了.

**Reply by dzqdzq** [reply]
感谢回复！ 希望能再完善一下这个Tree，非常感谢！

**Reply by dzqdzq** [reply]
我写的prefab解析插件也是准备开源的，不过我愿意付几杯咖啡钱，你们能帮忙解决这个问题。

**Comment by dzqdzq** [comment]
> 没有办法保证每个GitHub的提醒都及时看到, 也没有办法保证回答和解决所有问题. 我个人算是看消息比较积极的了, 当前 discussion 也确实首次看到, 邮件一提醒我就立马点进来了.
